src/main/java/uk/ac/ebi/embl/flatfile/writer/embl/EmblReducedFlatFileWriter.java

```diff
@@ -0,0 +1,106 @@
+import uk.ac.ebi.embl.api.entry.reference.Reference;
+import uk.ac.ebi.embl.api.validation.ValidationException;
+import uk.ac.ebi.embl.flatfile.EmblTag;
+import uk.ac.ebi.embl.flatfile.writer.EntryWriter;
+import uk.ac.ebi.embl.flatfile.writer.WrapType;
+import uk.ac.ebi.ena.taxonomy.taxon.Taxon;
+
+/** Reduced flat file writer.
+ */
+public class EmblReducedFlatFileWriter extends EntryWriter {
+
+    public static final String REDUCED_FF_WRITE_FAILED_MISSING_SEQUENCE = "Failed to write reduced flat file. Missing sequence for entry";
+    public static final String REDUCED_FF_WRITE_FAILED_MISSING_CONTIGS = "Failed to write reduced flat file. Missing contigs for CON entry";
+    public static final String REDUCED_FF_WRITE_FAILED_UNEXPECTED_CONTIGS = "Failed to write reduced flat file. Unexpected contigs for non-CON entry";
+
+    public EmblReducedFlatFileWriter(Entry entry) {
+        super(entry);
+        wrapType = WrapType.EMBL_WRAP;
+    }
+
+    private final boolean excludeSource = true;
+
+    public final static String SEPARATOR_LINE = EmblTag.XX_TAG  + "\n";
+    public final static String TERMINATOR_LINE = EmblTag.TERMINATOR_TAG  + "\n";
+
+    public boolean write(Writer writer) throws IOException {
+        if (entry == null) {
+            return false;
+        }
+
+        if (entry.getSequence() == null || entry.getSequence().getSequenceByte() == null) {
+            throw new IOException(REDUCED_FF_WRITE_FAILED_MISSING_SEQUENCE + ": " + entry.getPrimaryAccession());
+        }
+
+        if (entry.getDataClass() == Entry.CON_DATACLASS && !entry.isContigs()) {
+            throw new IOException(REDUCED_FF_WRITE_FAILED_MISSING_CONTIGS + ": " + entry.getPrimaryAccession());
+        }
+
+        if (entry.getDataClass() != Entry.CON_DATACLASS && entry.isContigs()) {
+            throw new IOException(REDUCED_FF_WRITE_FAILED_UNEXPECTED_CONTIGS + ": " + entry.getPrimaryAccession());
+        }
+
+        if(new IDWriter(entry).write(writer)) {
+            writer.write(SEPARATOR_LINE);
+        }
+
+        if (entry.getDataClass() == Entry.CON_DATACLASS) {
+
+            boolean expandedCON = isExpandedCON();
+            boolean notExpandedCON = isNotExpandedCON(); // in theory should be equal to !expandedCON
+
+            //Expanded entry: CO line, Feature, SQ line
+            if (expandedCON) {
+                new COWriter(entry, wrapType).write(writer);
+                writer.write(SEPARATOR_LINE);
+                writeFeatures(writer);
+                new EmblSequenceWriter(entry, entry.getSequence()).write(writer);
+            }
+
+            //Not expanded entry: Feature, CO line
+            if (notExpandedCON) {
+                writeFeatures(writer);
+                new COWriter(entry, wrapType).write(writer);
+            }
+        }
+        else {
+            writeFeatures(writer);
+            new EmblSequenceWriter(entry, entry.getSequence()).write(writer);
+        }
```

Comment:
expandedCon variable is not needed.
notExpandedCon variable is not needed.

Simplify code to:
```
// Always write sequence features.
writeFeatures(writer);

// Write CO lines for CONs.
if (entry.getDataClass() == Entry.CON_DATACLASS) {
  new COWriter(entry, wrapType).write(writer);
  writer.write(SEPARATOR_LINE);
}

// Always write sequence.
new EmblSequenceWriter(entry, entry.getSequence()).write(writer);
```